src/Controllers/HistoryViewerController.php

```diff
@@ -56,16 +64,88 @@ class HistoryViewerController extends LeftAndMain
             ];
         }
-
+        $clientConfig['endpoints'] = [
+            'revert' => $this->Link('api/revert'),
+            'read' => $this->Link('api/read'),
+        ];
         return $clientConfig;
     }
 
+    public function apiRead(HTTPRequest $request): HTTPResponse
+    {
+        $id = (int) $request->getVar('id');
+        $recordClass = $request->getVar('recordClass') ?? '';
+        $page = (int) $request->getVar('page');
+        if (!$page) {
+            $page = 1;
+        }
+        $obj = $this->getDataObject($id, $recordClass, true);
+        if (!$obj->canView()) {
+            $this->jsonError(403);
+        }
+        $list = Versioned::get_all_versions($recordClass, $obj->ID);
+        $totalCount = $list->Count();
+        $limit = HistoryViewerField::config()->get('default_page_size');
+        $offset = $limit * ($page - 1);
+        $list = $list->Sort('Version', 'DESC');
+        $list = $list->limit($limit, $offset);
+        $versions = [];
+        foreach ($list as $record) {
+            $author = $record->Author();
+            $publisher = $record->Publisher();
+            $versions[] = [
+                'version' => $record->Version,
+                'absoluteLink' => method_exists($record, 'AbsoluteLink') ? $record->AbsoluteLink() : '',
+                'author' => [
+                    'firstName' => $author ? $author->FirstName : '',
+                    'surname' => $author ? $author->Surname : '',
+                ],
+                'publisher' => [
+                    'firstName' => $publisher ? $publisher->FirstName : '',
+                    'surname' => $publisher ? $publisher->Surname : '',
+                ],
+                'deleted' => (bool) $record->WasDeleted,
+                'draft' => (bool) $record->WasDraft,
+                'published' => (bool) $record->WasPublished,
+                'liveVersion' => (bool) $record->isLiveVersion(),
+                'latestDraftVersion' => $record->isLatestDraftVersion(),
+                'lastEdited' => $record->LastEdited,
+            ];
+        }
+        $data = [
+            'pageInfo' => [
+                'totalCount' => $totalCount,
+            ],
+            'versions' => $versions,
+        ];
+        $this->extend('updateApiRead', $data, $request);
+        return $this->jsonSuccess(200, $data);
+    }
+
+    public function apiRevert(HTTPRequest $request): HTTPResponse
+    {
+        $this->checkJsonCsrfToken($request);
+        $id = (int) $this->getPostedJsonValue($request, 'id');
+        $recordClass = $this->getPostedJsonValue($request, 'recordClass');
+        $toVersion = (int) $this->getPostedJsonValue($request, 'toVersion');
+        $obj = $this->getDataObject($id, $recordClass, false);
+        if (!$obj->canEdit()) {
+            $this->jsonError(403);
+        }
+        if (!$obj->getAtVersion($toVersion)) {
+            $this->jsonError(400);
```

Comment:
If this is not 404 because the data was in a post body instead of in the "resource" (aka URL), add a code comment about that so we don't just swap it to 404 later on.

Reply:
It's a POST request so using 400